pkg/storage/storage.go

```diff
@@ -163,6 +164,42 @@ func New(cfg *config.Config) (*Storage, error) {
 		return tree.New()
 	}
 
+	// load the total memory of the server
+	vm, err := mem.VirtualMemory()
```

Comment:
Can you please add tests for this?

For tests we could add a way to set this limit arbitrarily, to something like 1 MB, and then make sure the system still works.

Reply:
already add the test for this PR